lib/fastlane/plugin/wpmreleasetoolkit/actions/common/buildkite_upload_pipeline_action.rb

```diff
@@ -0,0 +1,70 @@
+module Fastlane
+  module Actions
+    class BuildkiteUploadPipelineAction < Action
+      def self.run(params)
+        env_file = params[:env_file]
+        pipeline_file = params[:pipeline_file]
+        branch = params[:branch]
+        commit = params[:commit]
+
+        UI.user_error!("Pipeline file not found: #{pipeline_file}") unless File.exist?(pipeline_file)
+
+        UI.message "Uploading pipeline #{pipeline_file} on branch #{branch}, commit #{commit}"
+
+        ENV['BUILDKITE_BRANCH'] = branch
+        ENV['BUILDKITE_COMMIT'] = commit
+
+        if env_file && File.exist?(env_file)
+          UI.message("Sourcing environment file: #{env_file}")
```

Comment:
Nitpick, but I wonder if it might look odd to see in the logs "Uploading pipeline" first then "Sourcing environment file" next, while in practice the sourcing happens first and the upload happens next.

Maybe a simple fix for that is to update the wording to:
```suggestion
        UI.message "Uploading pipeline #{pipeline_file} on branch #{branch}, commit #{commit}"

        ENV['BUILDKITE_BRANCH'] = branch
        ENV['BUILDKITE_COMMIT'] = commit

        if pipeline_env_file && File.exist?(pipeline_env_file)
          UI.message(" - Sourcing environment file beforehand: #{pipeline_env_file}")
```

Reply:
Agreed! Updated together with other env var related changes here: 07661f76ab61bc32a665e8837c27c68a2a02c5f3